frontend/src/routes/login/login.tsx

```diff
@@ -0,0 +1,22 @@
+import LoginForm from "../../components/LoginForm/LoginForm";
+import styles from './login.module.css'
+import loginService from "../../services/login";
+import { LoginData } from "../../types";
+import { redirect } from "react-router";
+
+//@ts-ignore
```

Comment:
We can remove this line by declaring the types like so:

```tsx
export async function action({ request }: { request: Request }) {  
  const formData = await request.formData();
  const loginData = Object.fromEntries(formData) as unknown as LoginData;
  const response = await loginService.login(loginData);
  
  if (response) return redirect('/');
}
```

Reply:
Thank you! I became frustrated with it so I added the ts-ignore. Will change it!